scripts/main/build.js

```diff
@@ -143,17 +158,27 @@ build.album = function (data, disabled = false) {
 	return html;
 };
 
+/**
+ * @param {Photo}   data
+ * @param {boolean} disabled
+ *
+ * @returns {string} HTML for the photo
+ */
 build.photo = function (data, disabled = false) {
 	let html = "";
 	let thumbnail = "";
-	var thumb2x = "";
-	let isCover = data.id === album.json.cover_id;
-
-	let isVideo = data.type && data.type.indexOf("video") > -1;
-	let isRaw = data.type && data.type.indexOf("raw") > -1;
-	let isLivePhoto = data.livePhotoUrl !== "" && data.livePhotoUrl !== null;
-
-	if (data.sizeVariants.thumb === null) {
+	let thumb2x = "";
+	const isCover = data.id === album.json.cover_id;
+
+	const isVideo = data.type && data.type.indexOf("video") > -1;
+	const isRaw = data.type && data.type.indexOf("raw") > -1;
+	const isLivePhoto = data.live_photo_url !== "" && data.live_photo_url !== null;
+
+	// My IDE complains that `thumb` is always non-null and therefore the
+	// first branch of the condition is never taken.
+	// I am willing to agree.
+	// TODO: Revise this code, whether the first branch is really needed.
```

Comment:
The condition may not be coded correctly, but one of the cases it is meant to handle is that of videos with missing thumbnails (due to ffmpeg not being installed). That is something that definitely can happen and needs to be handled.

Reply:
Extremely good point. I have never thought about the case that we might be unable to extract a thumbnail for videos. In my world, the thumbnails always exists. I feat that this might even point to a bug on the server side, oh, oh ...